addons/web_editor/static/src/xml/editor.xml

```diff
@@ -148,8 +148,8 @@
 
             <div id="link" class="btn-group">
                 <div id="media-insert" title="Insert media" class="fa fa-file-image-o fa-fw btn editor-ignore"></div>
-                <div id="create-link" title="Insert or edit link" class="fa fa-link fa-fw btn editor-ignore"></div>
                 <div id="unlink" data-call="unlink" title="Remove link" class="fa fa-unlink fa-fw btn"></div>
+                <div id="create-link" title="Insert or edit link" class="fa fa-link fa-fw btn editor-ignore"></div>
```

Comment:
Fixing the bug indeed but this seems fragile, let's add a comment to indicate why this is important for this to be after unlink.

However, I'll also need confirmation from the editor team PO that those two buttons can be swapped... did you ask? It seems a bit inconsistent: in the link popover we have "edit" followed by "unlink":
![image](https://github.com/odoo/odoo/assets/10338094/05884c8a-0532-4ddf-8357-613f070f2b19)
But in the toolbar we'll now have:
![image](https://github.com/odoo/odoo/assets/10338094/4c463716-d493-4198-b93c-72b753e75d2d)
While before it was:
![image](https://github.com/odoo/odoo/assets/10338094/a24b6fa6-e596-400a-b7ff-078516770f32)

Reply:
In the end I kept the buttons at the same position in the editor